homeassistant/components/spotify/browse_media.py

```diff
@@ -265,80 +259,240 @@ def build_item_response(  # noqa: C901
     if media_content_type is None or media_content_id is None:
         return None
 
-    title = None
-    image = None
-    media: dict[str, Any] | None = None
-    items = []
+    title: str | None = None
+    image: str | None = None
+    items: list[ItemPayload] = []
 
     if media_content_type == BrowsableMedia.CURRENT_USER_PLAYLISTS:
-        if media := spotify.current_user_playlists(limit=BROWSE_LIMIT):
-            items = media.get("items", [])
+        if playlists := await spotify.get_playlists_for_current_user():
+            items = [
+                {
+                    "id": playlist.playlist_id,
+                    "name": playlist.name,
+                    "type": MediaType.PLAYLIST,
+                    "uri": playlist.uri,
+                    "thumbnail": playlist.images[0].url if playlist.images else None,
+                }
+                for playlist in playlists
+            ]
```

Comment:
Please deduplicate this, there are 16 almost identical copies of this list comprehension

Reply:
I want to definitely get the code complexity down of this function, but I didn't think this PR was the place to do that. Also because all the data we get is now typed, we can't just make it more generic without a lot of hassle